src/box-types.ts

```diff
@@ -0,0 +1,35 @@
+import {ReactNode, Component} from 'react'
+import * as CSS from 'csstype'
+
+type UIBoxProp = string | number | boolean | null | undefined
+export type CSSProps = CSS.StandardProperties<number | string | boolean>
+
+interface BoxPropsBase extends CSSProps {
+  is?: string | React.ComponentClass | React.FunctionComponent
+
+  className?: string
+
+  marginX?: UIBoxProp
+
+  marginY?: UIBoxProp
+
+  paddingX?: UIBoxProp
+
+  paddingY?: UIBoxProp
+
+  clearfix?: boolean
+
+  key?: string
+  /**
+   * Callback that gets passed a ref to inner DOM node (or component if the is prop is set to a
+   * React component type).
+   */
+  innerRef?(node: ReactNode): any
```

Comment:
I think the return type should be `void` here? https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#return-types-of-callbacks